### DISCUSSION TITLE
[Feature Request] Try to use a config file?

### DISCUSSION THREAD

**Discussion by ghost** [kRHYME7/Hyde-cli#4] [Ideas]
I was thinking, and i'd be happy to help on the implementation of this, i think it would be useful to have a config file for the configuration of Hyprdots-ctl. It doesn't need to have many options, the biggest reason i had in my mind was if we could change the URL and branch of where Hyprdots would be downloaded from without it being hard-coded into the package itself.

Some other ideas i had: 
 - Change how often it checks for updates
 - Specify the exact location of Hyprdots

All of this would be stored in something like `~/.config/hyprdots-ctl.conf`.
 
Also, i hope you don't mind if i use an issue page, i would start a discussion, but i don't think you enabled it. If you want to enable it that would be nice.

**Comment by kRHYME7** [comment]
>Change how often it checks for updates

Is 12 hours too soon right, What can you recommend? also The lock file wll be deleted after reboot so it would check updates every after boot

>Specify the exact location of Hyprdots

This might be very dangerous as all of the CLI will fail if It can't find the directory.  If you meant about the Fresh install I will add that option. 

but I did add the ``` Hyprdots --clone /path/to/dir ``` option but it is for testing, it won't be in ``` man``` for now. 

What can you recommend for how would this be implemented? 

I am testing now the control file system, I appended it with .conf.
This involves 1 script and a Control file, After restoring the default Hyprdots the script is run. 

I will try to use perl as perl is faster but I'm still trying to figure it out 
here is the WIP to restore the user Variables
https://github.com/kRHYME7/Hyprdots-ctl/blob/master/Scripts/define_user.sh

then here is the control file 
https://github.com/kRHYME7/Hyprdots-ctl/blob/master/Configs/hyprdots.conf

**Comment by kRHYME7** [comment]
> Also, i hope you don't mind if i use an issue page, i would start a discussion, but i don't think you enabled it. If you want to enable it that would be nice.

I'll try explore github now

**Comment by ghost** [comment]
> Is 12 hours too soon right, What can you recommend?

12 hours is fine, could be less too, in my opinion just don't bother the user if there is nothing to do.

> it won't be in ` man` for now

Perhaps we could add `Hyprdots man advanced` or something similar that's mostly for testing and could cause issues?

> What can you recommend for how would this be implemented?

I'm not really sure, i think that implementation is fine. 

Really, i just make this feature request because i think it'd be great to have more user control, you know? I also thought it'd be nice to maintain my fork of hyprdots with some personal modifications, and this script work with that also.

**Comment by kRHYME7** [comment]
Is the update query disturbing? like it take up time to load?

**Reply by ghost** [reply]
Sorry i forgot to respond to you earlier, what do you mean update query?

**Comment by kRHYME7** [comment]
Hi, @Skerse  I'll continue pondering about the control file. Can I have a list of the specific configs you want to be restored, I'm starting on the configs Hyprdots overwrites. Thanks in advance.

**Reply by ghost** [reply]
What else is there to restore besides what hyprdots overwrites? Unless it's possible to specify extra things to backup/restore, which would be great, but i don't expect you to create something like that since it's purely a hyprdots helper.

**Reply by kRHYME7** [reply]
> Unless it's possible to specify extra things to backup/restore, which would be great,

I slightly managed to do this more declaratively. ( I spaced it for clarity)

![image](https://github.com/kRHYME7/Hyprdots-ctl/assets/53417443/259541c2-20e2-4b28-96ae-a0f5d7b49e1e)

You can edit ``` ~/.config/hyprdots-ctl/manage_cfg.lst ``` to add extra files you want to manage(non existing source files will be skipped)

**Comment by kRHYME7** [comment]
> Specify the exact location of Hyprdots 

``` Hyprdots restore link /pth/to/Clone ``` Will link the specific directory as a Clone Directory.

**Comment by kRHYME7** [comment]
@Skerse Are you interested in making a discord server? DM me in case, ``` khing0v0 ```
